Update the product tours

cstns opened the pull request:
## Description

- removed unused tours (`device-tour` and `trial-tour`) and updated the welcome tour to fit the context of the new team home page
- extracted the animation logic from the scrollToAndJiggleHighlight UX composable to reuse in the last step of the new tour
- removed previous tour triggers from the applications page and moved them to the global team home page
- fixed the manual trigger for starting the tour to correctly redirect to the home page
- added a new pulse-style animation to highlight the editor button, as the existing jiggle effect was not effective on this tertiary-styled button
- adapted the `welcome-tour` that covers two main concepts (Hosted and Remote Instances), highlighting the at-a-glance stats and most recently modified instances for each, while also pointing users to dedicated pages for more options
- the tour additionally introduces the Recent Activity section, demonstrates the global search functionality (including keyboard shortcuts and navigation), and concludes with a clear CTA for the user to open the editor on their newly created instance

## Related Issue(s)

closes https://github.com/FlowFuse/flowfuse/issues/5741

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

joepavitt reviewed (changes requested):
<img width="677" height="325" alt="Screenshot 2025-07-14 at 10 28 55" src="https://github.com/user-attachments/assets/9d8e7697-3233-49bf-9852-55879439e4f2" />

The very last step doesn't align to the user's experience if there is not an Instance already present. In the self-hosted use case, that will not get created unless configured by the Platform admin.

Can we have a conditional that points them to "Create a Hosted Instance" instead if an Instance isn't available?

I've also pushed some trimming to the tour, as a lot of the home page and options are self-explanatory and don't need explicit call out.

cstns commented:
I split the last step into two and added some runtime logic to distinguish between cases where the user has hosted instances or not. It’s not perfect, but it works well enough.

I also fixed a bug in the manual tour trigger: previously, the team_slug was missing from the route when starting the tour from pages outside the team namespace.
